crates/router/src/core/payment_methods/vault.rs

```diff
@@ -286,6 +286,192 @@ impl Vaultable for api::PaymentMethodData {
     }
 }
 
+impl Vaultable for api::CardPayout {
+    fn get_value1(&self, _customer_id: Option<String>) -> CustomResult<String, errors::VaultError> {
+        let value1 = api::TokenizedCardValue1 {
+            card_number: self.card_number.peek().clone(),
```

Comment:
we need to refactor this `TokenizedCardValue1` struct, i agree this was added previously but it handles sensitive data and we are peeking it. let make its fields as secret and not use peek() here